src/selection.ts

```diff
@@ -14,6 +17,38 @@ export const getRange = (selection: Selection): Range => {
   return range;
 };
 
+export const cleanSelection = (selection: Selection): Selection => {
+  const anchor = selection.anchorNode;
+  const focus = selection.focusNode;
+
+  if (!anchor || !focus) {
+    return selection;
+  }
+
+  const anchorParent = anchor.parentNode;
+  const focusParent = focus.parentNode;
+  const beginsOnIframe = isIframe(anchor);
+  const endsOnIframe = isIframe(focus);
+  const range = getRange(selection);
+  const isImgAndFirstElement = isImg(range.startContainer);
+
+  // if selection starts w/ iframe or an img that is the first element of the page, replace anchorNode with its parent
+  // fixes firefox behavior that prevented starting highlights on iframes or images without previous siblings
+  const newAnchor = (beginsOnIframe || isImgAndFirstElement) && anchorParent ? anchorParent : anchor;
+
+  // if selection ends on an iframe, replace focusNode with its parent
+  // fixes firefox behavior that prevented ending highlights on iframes
+  const newFocus = endsOnIframe && focusParent ? focusParent : focus;
+
+  // if selection ends on an iframe, add 1 char to focus offset
+  // fixes firefox behavior that prevented ending highlights on iframes
+  const newFocusOffset = endsOnIframe && focusParent ? selection.focusOffset + 1 : selection.focusOffset;
```

Comment:
you tested this in safari and chrome right to make sure it doesn't break whatever they do?

Reply:
Yes, in fact it fixes an issue on Safari (highlighting only a video)